test/sprinkler_web/mqtt_test.exs

```diff
@@ -0,0 +1,28 @@
+defmodule SprinklerWeb.MqttTest do
```

Comment:
I think the whole file could be a test for the handler
```suggestion
defmodule SprinklerWeb.Mqtt.Handler do
```

Don't forget to change the name